### PR TITLE
[BCF - 3339] - Codec and CR hashed topics support

### PR DESCRIPTION
**Pull request by ilija42**
### Description

- Fix/implement support for filtering events in GetLatestValue that have indexed topics that get hashed
- Add comments that explain a bit better how codec encodes/decodes event topics
- Add TODOS with attached tickets for some improvements/bugs that I noticed

### CODE REVIEW

File: core/services/relay/evm/types/codec_entry.go [hunk 1]
```diff
@@ -200,7 +200,7 @@ func getNativeAndCheckedTypesForArg(arg *abi.Argument) (reflect.Type, reflect.Ty
 				return reflect.TypeOf(common.Hash{}), reflect.TypeOf(common.Hash{}), nil
 			}
 			fallthrough
-		case abi.SliceTy, abi.TupleTy, abi.FixedBytesTy, abi.FixedPointTy, abi.FunctionTy:
```
Comment (ilija42) [inline core/services/relay/evm/types/codec_entry.go:203]: FixedBytesTy don't get hashed so they can go to default case

File: core/services/relay/evm/event_binding.go [hunk 2]
```diff
@@ -340,11 +337,38 @@ func (e *eventBinding) derefTopics(topics []any) error {
 	return nil
 }
 
+// makeTopics encodes and hashes params filtering values to match onchain indexed topics.
+func (e *eventBinding) makeTopics(params []any) ([]common.Hash, error) {
+	// make topic value for non-fixed bytes array manually because geth MakeTopics doesn't support it
```
Comment (EasterTheBunny) [inline core/services/relay/evm/event_binding.go:342]: Is this a potential change we could propose for geth?

**Review by EasterTheBunny — APPROVED** [review]
LGTM

File: core/services/relay/evm/event_binding.go (hunk 2 above)
Comment (ilija42) [reply core/services/relay/evm/event_binding.go:342]: I thought about it, I'll probably raise a PR when I find some time

**Review by jmank88 — APPROVED** [review]

**Pull request merged**